src/ui/public/courier/saved_object/saved_object.js

```diff
@@ -231,59 +232,61 @@ export default function SavedObjectFactory(es, kbnIndex, Promise, Private, Notif
     };
 
     /**
-     * Save this object
-     *
-     * @return {Promise}
-     * @resolved {String} - The id of the doc
+     * Checks the id of the instance. If it is not set, creates a uuid for it.
      */
-    self.save = function () {
-
-      let body = self.serialize();
+    self.ensureId = function () {
+      this.id = this.id || uuid.v1();
 
-      // Slugify the object id
-      self.id = slugifyId(self.id);
-
-      // ensure that the docSource has the current self.id
+      // ensure that the docSource has the current id
       docSource.id(self.id);
-
-      // index the document
-      return self.saveSource(body);
     };
 
-    self.saveSource = function (source) {
-      let finish = function (id) {
-        self.id = id;
-        return es.indices.refresh({
-          index: kbnIndex
-        })
-        .then(function () {
-          return self.id;
-        });
-      };
+    /**
+     * If the object's title was changed, this changes the id so a save will trigger a
+     * 'Save a Copy'. Read https://github.com/elastic/kibana/issues/9056 and
+     * https://github.com/elastic/kibana/issues/9012 for some background into this issue.
+     * The goal is to move towards a better rename flow, but since our users have been conditioned
+     * to expect a 'save as' flow during a rename, we are keeping the logic the same until a better
+     * UI/UX can be worked out.
+     */
+    self.forceSaveAsOnTitleChange = function () {
+      if (this.titleChanged()) {
+        self.id = null;
+        self.ensureId();
+      }
+    };
 
-      return docSource.doCreate(source)
-      .then(finish)
-      .catch(function (err) {
-        // record exists, confirm overwriting
-        if (_.get(err, 'origError.status') === 409) {
-          let confirmMessage = 'Are you sure you want to overwrite ' + self.title + '?';
-
-          return safeConfirm(confirmMessage).then(
-            function () {
-              return docSource.doIndex(source).then(finish);
-            },
-            _.noop // if the user doesn't overwrite record, just swallow the error
-          );
-        }
-        return Promise.reject(err);
-      });
+    /**
+     * Returns true if the object's original title has been changed. New objects return false.
+     * @return {boolean}
+     */
+    self.titleChanged = function () {
```

Comment:
Minor nitpick: can we call this `isTitleChanged`? I really like the convention of prefixing flags and boolean-returning functions with `is`, `has`, `will`, `can`, etc. to reflect their role.